milli/src/criterion.rs

```diff
@@ -50,32 +50,95 @@ impl FromStr for Criterion {
             "sort" => Ok(Criterion::Sort),
             "exactness" => Ok(Criterion::Exactness),
             text => match AscDesc::from_str(text) {
-                Ok(AscDesc::Asc(field)) => Ok(Criterion::Asc(field)),
-                Ok(AscDesc::Desc(field)) => Ok(Criterion::Desc(field)),
+                Ok(AscDesc::Asc(Member::Field(field))) => Ok(Criterion::Asc(field)),
+                Ok(AscDesc::Desc(Member::Field(field))) => Ok(Criterion::Desc(field)),
+                Ok(AscDesc::Asc(Member::Geo(_))) | Ok(AscDesc::Desc(Member::Geo(_))) => {
+                    Err(UserError::InvalidRankingRuleName { name: text.to_string() })?
+                }
                 Err(UserError::InvalidAscDescSyntax { name }) => {
-                    Err(UserError::InvalidCriterionName { name }.into())
+                    Err(UserError::InvalidRankingRuleName { name }.into())
                 }
                 Err(error) => {
-                    Err(UserError::InvalidCriterionName { name: error.to_string() }.into())
+                    Err(UserError::InvalidRankingRuleName { name: error.to_string() }.into())
                 }
             },
         }
     }
 }
 
-#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
+pub enum Member {
+    Field(String),
+    Geo([f64; 2]),
+}
+
+impl FromStr for Member {
+    type Err = UserError;
+
+    fn from_str(text: &str) -> Result<Member, Self::Err> {
+        if let Some(point) = text.strip_prefix("_geoPoint(").and_then(|text| text.strip_suffix(")"))
+        {
+            let (lat, long) = point
+                .split_once(',')
+                .ok_or_else(|| UserError::InvalidRankingRuleName { name: text.to_string() })
+                .and_then(|(lat, long)| {
+                    lat.trim()
+                        .parse()
+                        .and_then(|lat| long.trim().parse().map(|long| (lat, long)))
+                        .map_err(|_| UserError::InvalidRankingRuleName { name: text.to_string() })
+                })?;
+            Ok(Member::Geo([lat, long]))
+        } else {
+            if is_reserved_keyword(text) {
+                return Err(UserError::InvalidReservedRankingRuleName { name: text.to_string() })?;
+            }
+            Ok(Member::Field(text.to_string()))
+        }
+    }
```

Comment:
I would prefer that you use a match instead ;)

```suggestion
        match text.strip_prefix("_geoPoint(").and_then(|text| text.strip_suffix(")")) {
            Some(point) => {
                let (lat, long) = point
                  .split_once(',')
                  .ok_or_else(|| UserError::InvalidRankingRuleName { name: text.to_string() })
                  .and_then(|(lat, long)| {
                      lat.trim()
                          .parse()
                          .and_then(|lat| long.trim().parse().map(|long| (lat, long)))
                          .map_err(|_| UserError::InvalidRankingRuleName { name: text.to_string() })
                  })?;
              Ok(Member::Geo([lat, long]))
        },
        None => {
            if is_reserved_keyword(text) {
                return Err(UserError::InvalidReservedRankingRuleName { name: text.to_string() })?;
            }
            Ok(Member::Field(text.to_string()))
        }
    }
```